src/charm_state.py

```diff
@@ -173,6 +174,7 @@ class SynapseConfig(BaseModel):  # pylint: disable=too-few-public-methods
     federation_domain_whitelist: str | None = Field(None)
     ip_range_whitelist: str | None = Field(None, regex=r"^[\.:,/\d]+\d+(?:,[:,\d]+)*$")
     public_baseurl: str | None = Field(None)
+    publish_rooms_allowlist: str | None = Field(None, regex=r"(@\w+:\w+\.\w+,?)+")
```

Comment:
Have yo considered treating this attr as a list here and parsing before setting it here?

Reply:
Excellent suggestion, thanks. Changed! Please, just check if what I did is the same as that you have in mind, I just added a validator to convert the item to list (and check the ids with the regex).

Reply:
I think that's actually better then what I had in mind :)